src/platform/qpg6100/qpg6100Config.cpp

```diff
@@ -46,106 +47,78 @@ CHIP_ERROR QPG6100Config::Init()
 
 uint16_t QPG6100Config::GetSettingsMaxValueLength(Key key)
 {
-    return qvCHIP_Nvm_GetMaxKeyLen(key);
+    uint16_t keyLen;
+
+    return (qvCHIP_Nvm_GetMaxKeyLen(key, &keyLen) == QV_STATUS_NO_ERROR) ? keyLen : 0;
 }
 
 CHIP_ERROR QPG6100Config::ReadConfigValue(Key key, bool & val)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
     uint16_t length;
-    bool res;
 
     length = sizeof(bool);
 
-    res = qvCHIP_Nvm_Restore(key, (uint8_t *) (&val), &length);
-    if (res == false)
-    {
-        err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
-    }
-
-    return err;
+    return MapNVMError(qvCHIP_Nvm_Restore(key, reinterpret_cast<uint8_t *>(&val), &length));
 }
 
 CHIP_ERROR QPG6100Config::ReadConfigValue(Key key, uint32_t & val)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
     uint16_t length;
-    bool res;
 
     length = sizeof(uint32_t);
 
-    res = qvCHIP_Nvm_Restore(key, (uint8_t *) (&val), &length);
-    if (res == false)
-    {
-        err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
-    }
-
-    return err;
+    return MapNVMError(qvCHIP_Nvm_Restore(key, reinterpret_cast<uint8_t *>(&val), &length));
 }
 
 CHIP_ERROR QPG6100Config::ReadConfigValue(Key key, uint64_t & val)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
     uint16_t length;
-    bool res;
 
     length = sizeof(uint64_t);
 
-    res = qvCHIP_Nvm_Restore(key, (uint8_t *) (&val), &length);
-    if (res == false)
-    {
-        err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
-    }
-
-    return err;
+    return MapNVMError(qvCHIP_Nvm_Restore(key, reinterpret_cast<uint8_t *>(&val), &length));
 }
 
 CHIP_ERROR QPG6100Config::ReadConfigValueStr(Key key, char * buf, size_t bufSize, size_t & outLen)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-    bool res;
+    qvStatus_t res;
     uint16_t length;
 
-    if (buf == NULL)
+    if ((buf == NULL) || (bufSize > USHRT_MAX))
```

Comment:
For portability, it would be better to use the explicit uint16 macro:

```suggestion
    if ((buf == NULL) || (bufSize > UINT16_MAX))
```